apiserver/facades/client/applicationoffers/applicationoffers.go

```diff
@@ -417,11 +431,28 @@ func (api *OffersAPI) GetConsumeDetails(args params.ApplicationURLs) (params.Con
 	}
 
 	for i, result := range offers.Results {
-		results[i].Offer = result.Result
+		offer := result.Result
+		results[i].Offer = offer
 		results[i].Error = result.Error
 		if result.Error == nil {
+			sourceModelTag, err := names.ParseModelTag(offer.SourceModelTag)
+			if err != nil {
+				results[i].Error = common.ServerError(err)
+				continue
+			}
 			results[i].ControllerInfo = controllerInfo
-			// TODO(Wallyworld) - add macaroon
+			// TODO(wallyworld) - wind back expiry time and add refresh
+			offerMacaroon, err := api.bakery.NewMacaroon("offer-"+offer.OfferName, nil,
```

Comment:
I don't think you want to specify a macaroon ID. Doing this means you'll only store one macaroon per offer. Assuming you want independent macaroons to be returned for GetConsumeDetails calls, just leave the ID empty, and the bakery service will generate a random one.

If you want it so that there's one per offer, then the refresh will become necessary so that you can revoke access to a specific user. You wouldn't be able to remove the macaroon from storage, because that would revoke access to *all* consumers.